### PR TITLE
add linting to test CI workflow, add task setup

### PR DESCRIPTION
**Pull request by davidbloss**
Add `task setup` as a generic local development task.
- This setups up `gofumpt` and `golangci-lint` locally and in CI pipelines
- This also serves as a place to add new tools when introduced

`task lint` added to CI

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -19,10 +19,17 @@ jobs:
...
+          version: 3.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
       - name: Run Tests
         run: |-
-          go test -race -coverprofile=coverage.out -covermode=atomic -v ./...
-          cat coverage.out | grep -v "/enum.go" > coverage.txt
+          task setup
+          task lint
+          task test
+          grep -v "/enum.go" coverage.out > coverage.txt
```
Comment (rocktavious) [inline .github/workflows/tests.yml:32]: any reason to no bake this into `task test` ?

Comment (davidbloss) [reply .github/workflows/tests.yml:32]: Not really :)
I can add it in

File: Taskfile.yml [hunk 2]
```diff
@@ -36,8 +25,43 @@ tasks:
...
+    cmds:
+      - task: install-go-tool
+        vars: { GO_TOOL: "gofumpt", GO_TOOL_PATH: "mvdan.cc/gofumpt@latest" }
+
+  install-golangci:
+    desc: go install "golangci-lint" and set GOBIN if not set
+    silent: true
+    cmds:
+      - task: install-go-tool
+        vars: { GO_TOOL: "golangci-lint", GO_TOOL_PATH: "github.com/golangci/golangci-lint/cmd/golangci-lint@v1.54.2" }
```
Comment (rocktavious) [inline Taskfile.yml:67]: can we add a comment to why the version is pinned vs using `@latest` ?

Comment (davidbloss) [reply Taskfile.yml]: I thought the golangci-lint page had a reason for pinning this, but I didn't see it. So I'm setting it to `@latest`

Comment (rocktavious) [reply Taskfile.yml]: 🎉

**Review by rocktavious — APPROVED** [review]

**Pull request merged**